core/src/main/java/nu/yona/server/analysis/service/ActivityService.java

```diff
@@ -310,6 +342,11 @@ private long getTotalPageableItems(UserAnonymizedDto userAnonymized, ChronoUnit
 				}));
 	}
 
+	private Interval limitInterval(Interval interval, LocalDate earliestPossibleDate)
```

Comment:
Can we not move this method to the `Interval` class?

Reply:
Yes, good idea.